query/src/main/java/org/infinispan/query/clustered/commandworkers/CQDelete.java

```diff
@@ -30,7 +33,8 @@ CompletionStage<QueryResponse> perform(BitSet segments) {
 
       SearchQueryBuilder query = queryDefinition.getSearchQueryBuilder();
       return blockingManager.supplyBlocking(() -> fetchIds(query), this)
-            .thenCompose(queryResult -> CompletionStages.performSequentially(queryResult.iterator(), cache::removeAsync,
+            .thenCompose(queryResult -> CompletionStages.performConcurrently(queryResult, 10,
+                  Schedulers.from(new WithinThreadExecutor()), cache::removeAsync,
```

Comment:
Does it make sense to cache `Schedulers.from(new WithinThreadExecutor())` and void instantiate a new one per query? 
Copy my comment from the chat
`I would suggest using statetransfer.chunk-size (default: 512) or lock.concurrency-level (default: 32)`

Reply:
+1 from me

Reply:
Honestly, I thought about it, but WithinThreadExecutor is pretty light weight.. I don't think caching is really going to be that helpful as the other costs around this will vastly outnumber it.

Using chunk size seems a bit too high to me... having 512 outstanding removes seems way too much. Using concurrency-level while is a bit odd, is probably okay without a better alternative.

Reply:
`availableCores() * 3` or times 4?

Reply:
Is fine by me. @fax4ever can tune it as needed afterwards imo.